inductiva/benchmarks/benchmark.py

```diff
@@ -243,3 +246,33 @@ def filter_distinct_columns(info):
         return filter_distinct_columns(info) \
             if columns == ColumnExportMode.DISTINCT \
             else info
+
+    def terminate(self) -> Self:
+        """
+        Terminates all active machine groups associated with the
+        benchmark.
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+
+        def _handle_suffix(executer):
+            if executer.host_type == ProviderType.GCP:
+                return "-".join(executer.vm_name.split("-")[:-1])
+            return executer.vm_name
+
+        machine_names = {
+            _handle_suffix(task.info.executer)
+            for task in self.get_tasks()
+            if task.info.executer
+        }
+
+        for machine in resources.get():
+            if machine.name not in machine_names:
+                continue
+            try:
+                machine.terminate(verbose=False)
+            except ApiException as api_exception:
+                logging.warning(api_exception)
```

Comment:
I tested the exception message. For example, if the ```terminate``` method raises an exception because it attempts to terminate an already terminated machine, it prints this message:

```
Attempting to terminate an unstarted machine group.
```

I think this is already quite nice -- the ```APIException``` class implements a custom ```__str__``` method.